### PR TITLE
Fixes #24723 - Supporting Rails 5.2

### PR DESCRIPTION
**Pull request by ik5**

### CODE REVIEW

**Comment by timogoebel** [comment]
This looks like a duplicate of #399.

**Comment by lzap** [comment]
Thanks, @ik5  and @ShimShtein are helping me to fix develop tests first, then I will pick one or another PR for the merge.

**Comment by lzap** [comment]
Discovery develop is finally green again, please rebase to retest on jenkins.

File: test/unit/discovery_taxonomy_extensions_test.rb [hunk 1]
```diff
@@ -1,6 +1,9 @@
 require 'test_plugin_helper'
 
 class DiscoveryTaxonomyExtensionsTest < ActiveSupport::TestCase
+  include FactImporterIsolation
+  allow_transactions_for_any_importer
```
Comment (lzap) [inline test/unit/discovery_taxonomy_extensions_test.rb:5]: Did you see failing tests or why this is needed? Let's probably add this in another PR then?

**Comment by lzap** [comment]
I am a bit confused, do we need both PRs or just one?

File: test/unit/discovery_taxonomy_extensions_test.rb (hunk 1 above)
Comment (ik5) [reply test/unit/discovery_taxonomy_extensions_test.rb:5]: yes you have failing tests, because you create facts on an open transaction

**Comment by ik5** [comment]
@lzap I need to remove one change here that @timogoebel made, and we are set, it's coming your way very soon :D

**Comment by ik5** [comment]
@lzap now it is ready :D

**Comment by lzap** [comment]
Thanks!

**Pull request merged**